jina/executors/indexers/vector.py

```diff
@@ -179,6 +191,19 @@ def ext2int_id(self) -> Optional[Dict]:
         if self.int2ext_id is not None:
             return {k: idx for idx, k in enumerate(self.int2ext_id)}
 
+    def update(self, keys: Iterator[int], values: Iterator[bytes], *args, **kwargs):
+        self.delete(keys)
+        self.add(np.array(keys), np.array(values))
+
+    def delete(self, keys: Iterator[int], *args, **kwargs):
+        # TODO check if they exist
```

Comment:
Should this be part of this PR already?

Reply:
Oh right. I'll add that and a test for it.